metricbeat/docs/developer-guide/create-metricset.asciidoc

```diff
@@ -146,18 +141,19 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 [float]
 ==== Fetching
 
-The `Fetch` method is the central part of the MetricSet. Fetch is called every
-time new data is retrieved. In case more then one host is defined, fetch is
-called for each host once. The frequency of calling fetch is based on the `period`
+The `Fetch` method is the central part of the metricset. `Fetch` is called every
+time new data is retrieved. If more than one host is defined, `Fetch` is
+called once for each host. The frequency of calling `Fetch` is based on the `period`
 defined in the configuration file.
 
-Fetch must return a `common.MapStr` object which is then sent to elasticsearch.
-In case an error happens, an error must be returned which is then sent instead
-to elasticsearch. That means metricbeat always sends an event, also on failure.
-It must made sure, that the error message helps to identify what the actual
-error is.
+`Fetch` must return a `common.MapStr` object, which is then sent to Elasticsearch.
+If an error happens, the error must be returned and then sent instead
+to Elasticsearch. This means that Metricbeat always sends an event, even on failure.
+You must make sure that the error message helps to identify the actual error.
+
+The following example shows a counter that is incremented by one for each `Fetch` call:
```

Comment:
The following example shows a metricset Fetch method with a counter which is incremented for each `Fetch` call.